QUARKUS-6259 - Enable named data sources for Hibernate Reactive (#255)
quarkus-qe/quarkus-test-plans: QUARKUS-6259.md

# QUARKUS-6245 Enable named data sources for Hibernate Reactive

JIRA:  https://issues.redhat.com/browse/QUARKUS-6259

Upstream PRs:
    - https://github.com/quarkusio/quarkus/pull/47631
    - https://github.com/quarkusio/quarkus/pull/49287

Documentation:
    - https://quarkus.io/version/main/guides/hibernate-reactive#multiple-persistence-units

This feature allow to configure multiple (named) datasources with hibernate-reactive extension.

> By extend it allow named persistence units.

Author: We agreed with @rsvoboda that I will do tests for both named DS + PU. There is no JIRA for https://github.com/quarkusio/quarkus/pull/48007 but it is going to be.

> By extend it allow named persistence units.

Reviewer: Ok, as you have linked the issue and plan to test it, I agree.

## Scope of testing
Tests will cover basic working with an app, that is connected to multiple databases.

> There will be 3 databases used in total - two of same type (e.g. postgres) and one different. 

Reviewer: If our test executors can handle 3 DBs and 2 of them are same (like PG) so at most 2 images are downloaded, I am fine with keeping 3. I think it will be a lot, but since it is very easy to rewrite it to 2 PG PUs against one DB instance, I think we can try it and see...

> Goal is to verify if Quarkus with hibernate-reactive correctly handles multiple types of DB at once and if same driver can work with multiple DBs simultaneously.

Reviewer: I am fine with Fedor's sentence that was applied, so resolving this comment thread. (even though that hibernate-reactive is still there, np)

As part of this testing, every DB will be initiated with separate sql-load-script to verify this works properly.

## Existing test coverage
Upstream tests were added as part of https://github.com/quarkusio/quarkus/pull/47631, but these cover only work with one named datasource.
Which is not sufficient, as we can have multiple datasources.
Our testsuite already has a significant coverage for hibernate in module `sql-db/hibernate-reactive`.

> If possible tests for this feature should reuse this test coverage, only change underlying app to use multiple named datasources.

Author: > Also I kinda wonder if this sentence doesn't belong to the "Automated test development" section.
Kind of does, but it also depend on existing test coverage.

Original idea was to implement it into the existing module, but I'm not sure how much will I need to change the underlying app. And if the complexity is too much, it might not be worth it to basically combine two apps into one.
But as you stated below, as I can just load only specific classes, so I will rephrase this, so it will be in this module.

## Automated test development
New test class will be added into module `sql-db/hibernate-reactive`, 

> if this it is possible to implement named datasources there, without disrupting existing tests.
> If named datasources prove to be too disruptive, new module will be created.

Reviewer 2: ```suggestion
named datasources should be implemented there, unless they disrupt existing tests.
Otherwise, a new module needs to be created.
```

Author: I omitted this entire sentenses, so closing this one.

## Impact on TS and resources
New test class will be created, running on JVM and native on both bare metal and OCP.

> On bare metal JVM run will take about 1 minute and native 3 minutes. 

Reviewer 2: In addition to what Michal said, I see no assumptions for native. No need to be precise, but something like "native runs are expected to be 5 time longer" would be handy.
